src/components/ShareList.jsx

```diff
@@ -0,0 +1,99 @@
+import { shareList } from '../api/firebase';
+import { useState } from 'react';
+import { useAuth } from '../api';
+import { validateEmail } from '../utils/emailValidator.js';
+
+export function ShareList({ listPath }) {
+	const [recipientEmail, setRecipientEmail] = useState('');
+	const [emailValidator, setEmailValidator] = useState('');
+	const [isValid, setIsValid] = useState(false);
+	const { user } = useAuth();
+	const currentUserId = user?.uid;
+
+	const handleChange = (e) => {
+		setRecipientEmail(e.target.value);
+
+		if (validateEmail(e.target.value)) {
+			setEmailValidator('');
+			setIsValid(true);
+		} else {
+			setEmailValidator(
+				'Please enter a valid email, e.g., example@example.com.',
+			);
+		}
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		if (recipientEmail === '') {
+			return alert('Cannot submit without an email.');
+		}
+
+		if (!isValid) {
+			return alert('Cannot submit with an invalid email');
+		}
+
+		try {
+			await shareList(listPath, currentUserId, recipientEmail);
+			alert('List was successfully shared!');
+			setRecipientEmail('');
+		} catch (error) {
+			alert(error);
+		}
+	};
+
+	return (
+		<form onSubmit={handleSubmit}>
+			<p>You can share this shopping list with existing users!</p>
+			<label htmlFor="shareEmail">Enter recipient's email:</label>
+			<input
+				id="shareEmail"
+				type="text"
+				value={recipientEmail}
+				required
+				onChange={handleChange}
+			/>
+			<p>{emailValidator}</p>
+			<button type="submit">Submit</button>
+		</form>
+	);
+}
+
+// We want to further refactor our validators. Below is some code we would like to further refactor and impliment at a later date.
+
```

Comment:
This is really interesting. I think it's worth discussing as a team. Solution 2,  has some features that may make it more accessible and user-friendly. Dynamically updating the validation messages on the screen allows users to use screen readers, as it is part of the DOM.  The alerts you have in solution 1, aren't as accessible to screen readers BUT since your team is considering implementing a different solution for alerts at the end of the project it might not be an issue you have to account for right now.  

I think it boils down to the user experience you want to provide. Some users may dislike seeing a message appear while they are typing, while others may enjoy it. 

Since you've used alerts in other issues, I suggest going with option 1 for now and discussing the solution as a team. Decide whether you'll want to create/import different types of alerts, render the text conditionally as a `<p>` tag, or use dynamic rendering throughout the app. Then, create issues for the final weeks of the project and plan on implementing them at that time.

Reply:
Perfect, thanks Christina! This is exactly what Paige and I were thinking.